app/controllers/maps_controller.rb

```diff
@@ -133,6 +152,48 @@ def events
     end
   end
 
+  # GET maps/:id/approve_access/:request_id
+  def approve_access
+    request = AccessRequest.find_by_id(params[:request_id])
```

Comment:
Hmm, kk. But, the reason that I went with this was specifically because AccessRequest.find throws an error when it doesn't find one, which i don't want. Hmm or maybe I do. I was thinking of just redirecting them and displaying the message as a nicer option, but maybe it's unnecessary, given that the times when someone would hit this URL and not find one is incrediblly rare, they should only ever hit this url by clicking through "Accept" or "Deny" links out of an email...